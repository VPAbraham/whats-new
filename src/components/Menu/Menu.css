nav {
  color: white;
  float: left;
  font-family: 'Poppins', sans-serif;
  height: 100vh;
  width: 20%;
  display: flex;
  flex-direction: column;
}

.nav-div {
  display: flex;
  align-items: center;
  margin: 50px 15px 0 30px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
}

.nav-div :hover{
  box-shadow: 0 14px 28px rgba(35, 34, 34, 0.25), 0 10px 10px rgba(23, 22, 22, 0.22);
}

.nav-div p {
  font-size: 1.5em;

}


.icon {
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 15px;
}


