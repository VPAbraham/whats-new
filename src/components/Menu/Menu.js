import React, { Component } from 'react';
import './Menu.css'
import NewsIcon from './menu-images/newspaper.png'
import TechIcon from './menu-images/data.png'
import EntIcon from './menu-images/video-camera.png'
import SciIcon from './menu-images/laboratory.png'
import HealthIcon from './menu-images/care.png'

const icons = {
  'Local': NewsIcon,
  'Technology': TechIcon,
  'Entertainment': EntIcon,
  'Science': SciIcon,
  'Health': HealthIcon
}

const MenuItem = (props) => {
  const { topic, changeTopic } = props
  return (
    <div onClick={() => changeTopic(topic)}>
      <img src={icons[topic]}/>
      <p>{topic}</p>
    </div>
  )
}

class Menu extends Component{
  constructor(props) {
    super()
  }

  render() {
    console.log(this.props.topics)
    let mappedMenuItems = Object.keys(this.props.topics).map(topic => {
      return <MenuItem topic={topic} changeTopic={this.props.changeTopic}/>
    })
    
    return (
      <nav className="nav">
        {mappedMenuItems}
      </nav>
    )
  }
}

export default Menu;