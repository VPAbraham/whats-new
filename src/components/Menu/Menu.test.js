import React from 'react'
import { shallow } from 'enzyme'
import Menu from './Menu'
import Local from '../../data/local'
import Technology from '../../data/technology'
import Entertainment from '../../data/entertainment'
import Science from '../../data/science'
import Health from '../../data/health'

describe('Menu', () => {
  let wrapper
  let mockChangeTopic = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Menu 
    topics={
      Local,
      Technology,
      Entertainment,
      Science,
      Health
    }
    changeTopic = {mockChangeTopic}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})