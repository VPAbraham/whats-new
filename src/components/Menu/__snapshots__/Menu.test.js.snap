// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<nav
  className="nav"
>
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="0"
    topic="0"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="1"
    topic="1"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="2"
    topic="2"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="3"
    topic="3"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="4"
    topic="4"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="5"
    topic="5"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="6"
    topic="6"
  />
  <MenuItem
    changeTopic={[MockFunction]}
    className="nav-div"
    key="7"
    topic="7"
  />
</nav>
`;
